src/service/translate.js

```diff
@@ -1780,10 +1780,10 @@ function $translate($STORAGE_KEY, $windowProvider, $translateSanitizationProvide
           // we are already loading this asynchronously
           // resolve our new deferred when the old langPromise is resolved
           langPromises[key].then(function (translation) {
-            if (!$uses) {
-              useLanguage(translation.key);
+            if ($nextLang === key) {
```

Comment:
That is what I meant with the last comment: I first thought, that there might not be a difference in between key and translation.key , but due to a very slow reply, this could possibly happen after rethinking... or not?